### ISSUE TITLE
bug: `go mod` fails if I remove `rsc.io/quote` import in new go script

### ISSUE THREAD

**Issue by komish** [windmill-labs/windmill#688]
### Describe the bug

Note: I'm running this on rootless podman (not docker). Initial tests with demo scripts have worked, but once I try to manipulate the script a bit, I'm running into issues. Not enough information to debug in the logs that I can see. 

```
2022-10-05T20:52:10.880722Z  INFO windmill::worker: fetched job 0183a9eb-8f74-1b4b-15cf-016f0f4b79a1 worker=dt-worker-4CDDY-TVm6M id=0183a9eb-8f74-1b4b-15cf-016f0f4b79a1
2022-10-05T20:52:10.880808Z  INFO windmill::worker: handling job 0183a9eb-8f74-1b4b-15cf-016f0f4b79a1 worker=dt-worker-4CDDY-TVm6M job_id=0183a9eb-8f74-1b4b-15cf-016f0f4b79a1 workspace_id=homelab
2022-10-05T20:52:10.881234Z  INFO windmill::worker: started Go job 0183a9eb-8f74-1b4b-15cf-016f0f4b79a1 worker_name=dt-worker-4CDDY-TVm6M job_id=0183a9eb-8f74-1b4b-15cf-016f0f4b79a1 workspace_id=homelab is_ok=false
2022-10-05T20:52:10.951254Z  INFO windmill::worker: go mod error
```

### To reproduce

1. Create a go script
2. Modify the script to look like below

```go
import (
	"fmt"
	// "rsc.io/quote"
)

func main(user string) (string, error) {
  // fmt.Println(quote.Opt())
  return fmt.Sprintf("Hello, %s!", user), nil
}
```

4. Hit test
6. See error

```
job 0183a9eb-8f74-1b4b-15cf-016f0f4b79a1 on worker dt-worker-4CDDY-TVm6M


--- GO DEPENDENCIES SETUP---

go: creating new go.mod: module mymod
go: to add module requirements and sums:
	go mod tidy
Lock file generation failed. Status: Err(ExecutionErr("job process killed"))
```

### Expected behavior

It should run (and then subsequently deploy). I can seemingly add some other imports (e.g. `strings` seems to import without a problem).

### Application version

v1.36.0-40-gc8c0daf

### Additional Context

Using the container image at `:main` --> `ghcr.io/windmill-labs/windmill@sha256:dfef65413a8a17055adef9e4872e1ed821d2c8d0f9c1f5737f810fbc71227f1f`

Podman  `podman version 4.2.1`

**Comment by rubenfiszel** [comment]
Thanks for the report, additional question, does this happen all the time  with those same imports or just some of the time ? (One way to test this is to preview a few times)

Otherwise it's related to #641 which I suspect to be an OOM issue due to nsjail constraints. I will investigate promptly

**Comment by komish** [comment]
@rubenfiszel It seems to happen every time with the provided script above. I'm not sure what you mean by preview, but if I hit "Test" on the script in the editor, it happens constantly. If I save, it fails to deploy.

One thing to note is that I forgot to deploy the LSP. I don't think it matters, but can't hurt to mention.

Podman command to launch the app looks like this. I've also got DISABLE_NSJAIL set to true. Otherwise, it's pretty much identical to the self-hosted documentation. 

```
podman run \
    -d \
    --name windmill-app \
    -e DATABASE_URL=postgres://postgres:supersecret@localhost/windmill?sslmode=disable \
    -e BASE_URL=https://subdomain.example.com \
    -e BASE_INTERNAL_URL=http://localhost:8000 \
    -e RUST_LOG=info \
    -e NUM_WORKERS=3 \
    -e RUST_BACKTRACE=1 \
    -e DISABLE_NUSER=false \
    -e DENO_PATH=/usr/bin/deno \
    -e PYTHON_PATH=/usr/local/bin/python3 \
    -e DISABLE_NSJAIL=true \
    -e NSJAIL_PATH=nsjail \
    --pod windmill-test \
    ghcr.io/windmill-labs/windmill:main
```

**Comment by rubenfiszel** [comment]
Could you share what operating systems this is running on ? I can reproduce on app.windmill.dev that runs on Rocky Linux 6 but I cannot reproduce locally, even using your podman. Will investigate further, very frustrating.

**Comment by rubenfiszel** [comment]
Thanks for the feedback. I will test more on our servers, I want to get to the bottom of this, it's one of the most surprising bug I've seen. The process running "go mod tidy" seem to get killed for some reasons. I have tested running windmill with low memory but it doesn't fail the same way so it's not just OOM. The "go mod tidy" is done outside of nsjail so it cannot be related to nsjail. The SELinux is a good path to investigate.

**Comment by rubenfiszel** [comment]
@komish Would you have the opportunity to test with: feb5aad, the latest on main ? It fixed the issue on app.windmill.dev

**Comment by komish** [comment]
Looks like I can now test without issue! But I can't seem to deploy. Once I hit save on the go script, I received this message:

```
Error deploying this script

This script has not been deployed successfully because of the following errors:

job 0183b2b2-102e-7649-4c4c-d02fc2d3e79f on worker dt-worker-Y8UVs-tGwLM

go: creating new go.mod: module mymod
go: to add module requirements and sums:
	go mod tidy
IO error: No such file or directory (os error 2)
```

Script looks like this:

```go
import (
	"fmt"
	// "rsc.io/quote"
  "strings"
)

func main(x string) (interface{}, error) {
	fmt.Println("Hello, World")
	// fmt.Println(quote.Opt())
  return strings.Join([]string{"a", x, "b"}, "-"), nil
}
```

Happy to open a new issue if this is unrelated.

**Comment by komish** [comment]
The image that's deployed is as follows:

```
$ podman image inspect ghcr.io/windmill-labs/windmill:main | jq .[].Digest -r
sha256:7fb64718c8a3a7bb5cf932b84d5433322f98030dce7366dfe4a18b613829a403
```

**Comment by rubenfiszel** [comment]
Yes, locking dependencies for go is slightly broken now, I'm actively working on it. Thanks for the report!

**Comment by rubenfiszel** [comment]
Fixed on  [22c4a3b](https://github.com/windmill-labs/windmill/commit/22c4a3b37574b7dfab7dde0420dd40235acec350)

**Comment by komish** [comment]
Confirmed. Closing this. Thanks for everything @rubenfiszel!
